core/src/main/java/io/aiven/kafka/tieredstorage/chunkmanager/DiskBasedCachingChunkManager.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tieredstorage.chunkmanager;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.concurrent.Executor;
+
+import com.github.benmanes.caffeine.cache.RemovalListener;
+import com.github.benmanes.caffeine.cache.Weigher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.nio.file.StandardCopyOption.ATOMIC_MOVE;
+
+public class DiskBasedCachingChunkManager extends CachingChunkManager<Path> {
+
+    private static final Logger log = LoggerFactory.getLogger(DiskBasedCachingChunkManager.class);
+    static final String TEMP_CACHE_DIRECTORY = "temp";
```

Comment:
nit: `TMP` instead of `TEMP` seem to be more broadly used.
```suggestion
    static final String CACHE_TMP_DIRECTORY = "tmp";
```

Reply:
Reconsidering this, as `temp` already used in our code base.